.github/workflows/build.yml

```diff
@@ -0,0 +1,149 @@
+name: Build
+env:
+  GEOS_VERSION: "3.6.5"
+
+on:
+  push:
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+      - "packages/basemap_data/**"
+      - "packages/basemap_data_hires/**"
+  pull_request:
+    paths:
+      - ".github/workflows/**"
+      - "packages/basemap/**"
+      - "packages/basemap_data/**"
+      - "packages/basemap_data_hires/**"
+  workflow_dispatch:
+
+jobs:
+  build_data:
+    name: Build data packages
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        package: [basemap_data, basemap_data_hires]
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.8"
+
+      - name: Install build dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install build wheel
+
+      - name: Build package
+        run: |
+          cd packages/${{ matrix.package }}
+          python -m build --wheel --sdist
+
+      - uses: actions/upload-artifact@v4
+        with:
+          path: packages/${{ matrix.package }}/dist/*
+          name: dist-${{ matrix.package }}
+
+  build_basemap:
+    name: Build basemap wheels (${{ matrix.os }})
+    needs: [build_data]
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest]
+        include:
+          - os: ubuntu-latest
+            before_all: |
+              echo "Starting BEFORE_ALL script"
+              cd "{package}"
+              export GEOS_DIR="$(pwd)/extern"
+              echo "GEOS_DIR set to: $GEOS_DIR"
+              python -c "import sys; sys.path.append('utils'); from GeosLibrary import GeosLibrary; GeosLibrary('3.6.5', root='extern').build('extern', njobs=16)"
+            environment: >-
+              CFLAGS="-I/project/packages/basemap/extern/include -I/project/packages/basemap/extern/include/geos"
+              LDFLAGS="-L/project/packages/basemap/extern/lib"
+              PIP_DEFAULT_TIMEOUT=10
+              PIP_DISABLE_PIP_VERSION_CHECK=1
+              PIP_PREFER_BINARY=1
+              PIP_RETRIES=0
+              PYTHONUNBUFFERED=1
+              PYTHONWARNINGS="ignore:DEPRECATION"
+
+          - os: windows-latest
+            before_all: >-
+              pwsh -Command "
+              Write-Host 'Starting BEFORE_ALL script';
+              $pwd = Get-Location;
+              Write-Host 'Current directory:' $pwd;
+              Set-Location '{package}';
+              $env:GEOS_DIR = '{package}/extern';
+              python -c 'import sys, os; sys.path.append(\"utils\"); from GeosLibrary import GeosLibrary; GeosLibrary(\"3.6.5\").build(\"extern\", njobs=16)'
+              "
+            environment: >-
+              GEOS_DIR="D:/a/basemap/basemap/packages/basemap/extern"
+              CFLAGS="-ID:/a/basemap/basemap/packages/basemap/extern/include -ID:/a/basemap/basemap/packages/basemap/extern/include/geos/include"
```

Comment:
This is just a comment for the future: I would like to see if `CFLAGS` and `LDFLAGS` can be removed from here at some point *after* this PR. Looking at the logs of the Windows builds, it can be seen that `GEOS_DIR` is identified successfully by the `setup.py` file, so `CFLAGS` and `LDFLAGS` are set correctly by `setup.py` without the need to override them manually.

Reply:
Hmm that is strange. I have looked at what the utils directory is doing but in my runs it wasn't picking it up without setting this manually.

Reply:
Resolving here because it was safe to remove `CFLAGS` and `LDFLAGS` after setting `GEOS_DIR`.